src/framework/theme/components/select/select.component.ts

```diff
@@ -1156,9 +1156,14 @@ export class NbSelectComponent implements OnChanges, AfterViewInit, AfterContent
    * */
   protected setSelection(value) {
     const isArray: boolean = Array.isArray(value);
+    let formattedValue = value;
 
-    if (this.multiple && !isArray) {
-      throw new Error('Can\'t assign single value if select is marked as multiple');
+    if (this.multiple) {
+      if (value === null || value === undefined) {
+        formattedValue = [];
+      } else if (!isArray) {
+        throw new Error('Can\'t assign single value if select is marked as multiple');
+      }
     }
```

Comment:
Too much nesting. Let's move `null` to empty array coercion to the line right after `safeValue` declaration and add one more flag value like `isResetValue` to the `if (this.multiple && !isArray)`, something like:
```
const isResetValue = value == null;
let safeValue = value;
if (this.multiple) {
  safeValue = value ?? [];
}

if (this.multiple && !isArray && !isResetValue) {
  throw ...
```
Do you agree?

BTW, to check for both `null` and `undefined` we use loose equality comparison with `null`, it's shorter:
```value == null```